editor/src/components/canvas/remix/remix-utils.tsx

```diff
@@ -99,3 +128,397 @@ export function getRoutesFromRouteManifest(
     return []
   }
 }
+
+export function createAssetsManifest(routes: RouteManifest<EntryRoute>): AssetsManifest {
+  return {
+    entry: { imports: [], module: '' },
+    url: '/',
+    version: '1',
+    routes: routes,
+  }
+}
+
+export interface RouteModuleCreator {
+  filePath: string
+  executionScopeCreator: (projectContents: ProjectContentTreeRoot) => ExecutionScope
+}
+
+export interface RouteIdsToModuleCreators {
+  [routeId: string]: RouteModuleCreator
+}
+
+export interface RouteModulesWithRelativePaths {
+  [routeId: string]: {
+    relativePath: ElementPath
+    filePath: string
+  }
+}
+
+export interface GetRoutesAndModulesFromManifestResult {
+  routeModuleCreators: RouteIdsToModuleCreators
+  routes: Array<DataRouteObject>
+  routeModulesToRelativePaths: RouteModulesWithRelativePaths
+  routingTable: RemixStaticRoutingTable
+}
+
+export function getDefaultExportNameAndUidFromFile(
+  projectContents: ProjectContentTreeRoot,
+  filePath: string,
+): { name: string; uid: string } | null {
+  const file = getProjectFileByFilePath(projectContents, filePath)
+  if (file == null || file.type != 'TEXT_FILE' || file.lastParseSuccess == null) {
+    return null
+  }
+
+  const defaultExportName =
+    file.lastParseSuccess.exportsDetail.find(
+      (e): e is ExportDefaultFunctionOrClass => e.type === 'EXPORT_DEFAULT_FUNCTION_OR_CLASS',
```

Comment:
We could do with one of our `isX` functions here as then it would be just:
```suggestion
      isExportDefaultFunctionOrClass
```